### PR TITLE
doc: update the upgrade documentation to reflect 3.4.0 changes

### PR DESCRIPTION
**Pull request by pkalever**
# Describe what this PR does #

* Mainly removed rbd-nbd mounter specified at the pre-upgrade
considerations affecting the restarts.

* Also updated the 3.3 tags to 3.4

Signed-off-by: Prasanna Kumar Kalever \<prasanna.kalever@redhat.com\>

### CODE REVIEW

**Review by Madhu-1 — DISMISSED** [review]
Handling node reboot hangs due to existing network mounts can also be removed and looks like we need to make a statement no direct upgrade from 3.x except 3.3 to 3.4

**Comment by Madhu-1** [comment]
This needs to be backported to 3.4 branch as its targeting devel branch

**Review by yati1998 — APPROVED** [review]

**Comment by pkalever** [comment]
> Handling node reboot hangs due to existing network mounts can also be removed and looks like we need to make a statement no direct upgrade from 3.x except 3.3 to 3.4

Updated PR with these too, @Madhu-1 PTAL. Thanks!

**Review by Madhu-1 — APPROVED** [review]

**Comment by Madhu-1** [comment]
> > Handling node reboot hangs due to existing network mounts can also be removed and looks like we need to make a statement no direct upgrade from 3.x except 3.3 to 3.4
> 
> Updated PR with these too, @Madhu-1 PTAL. Thanks!

Thanks @pkalever

File: docs/ceph-csi-upgrade.md [hunk 1]
```diff
@@ -107,12 +108,15 @@ that will not be supported in the official releases. Builds from the devel
 branch can have functionality changed and even removed at any time without
 compatibility support and without prior notice.
 
-git checkout v3.3.0 tag
+**Also, we do not recommend any direct upgrades to 3.4 except from 3.3 to 3.4.**
```
Comment (Yuggupta27) [inline docs/ceph-csi-upgrade.md:111]:
```suggestion
**Also, we recommend upgrading by only one release at a time.**
```

**Review by Yuggupta27 — CHANGES_REQUESTED** [review]
Instead of keeping it specific to the v3.4 release, This general comment can be added throughout the doc at the top. What do you think?

File: docs/ceph-csi-upgrade.md (hunk 1 above)
Comment (Yuggupta27) [reply docs/ceph-csi-upgrade.md:111]: And also, you can include the example as you have already done :+1:

Comment (pkalever) [reply docs/ceph-csi-upgrade.md:111]: I will take it as a nit, if this PR needs any respin further, I will consider clubbing it along, else we can take it later.

**Comment by pkalever** [comment]
> Instead of keeping it specific to the v3.4 release, This general comment can be added throughout the doc at the top. What do you think?

@Yuggupta27 
I don't see a better fit, the current location looks fine to me.

**Comment by pkalever** [comment]
@Yuggupta27 see if you have missed this line in the  pre-upgrade considerations section

```
This guide will walk you through the steps to upgrade the software in a cluster
from v3.3 to v3.4
```

So the focus of this doc will be about the current release always.

**Comment by Yuggupta27** [comment]
> @Yuggupta27 see if you have missed this line in the pre-upgrade considerations section
> 
> ```
> This guide will walk you through the steps to upgrade the software in a cluster
> from v3.3 to v3.4
> ```
> 
> So the focus of this doc will be about the current release always.

Sure, I was referring to the whole document `ceph-csi-upgrade.md` and not just the section `upgrade the software in a cluster
from v3.3 to v3.4`.
Anyways just a nit, we can proceed and do it later ;) .

**Review by Yuggupta27 — APPROVED** [review]

**Review by humblec — APPROVED** [review]

**Comment by humblec** [comment]
Thanks @pkalever :+1: , much appreciated !

**Pull request merged**